Generate better ChangeLogs in update-angle script

kenrussell opened the pull request:
#### 83c12bcc9d4e1cae1d705493356909e21e90d905
<pre>
Generate better ChangeLogs in update-angle script
<a href="https://bugs.webkit.org/show_bug.cgi?id=238649">https://bugs.webkit.org/show_bug.cgi?id=238649</a>

Patch by Kenneth Russell &lt;kbr@chromium.org &gt; on 2022-05-25
Auto-generate the preferred commit message format during ANGLE rolls
into a commit-message.txt file.

Reviewed by Kimmo Kinnunen.

* Tools/Scripts/update-angle:

Canonical link: <a href="https://commits.webkit.org/251002@main">https://commits.webkit.org/251002@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@294872">https://svn.webkit.org/repository/webkit/trunk@294872</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

kkinnunen-apple commented on Tools/Scripts/update-angle:
```diff
@@ -121,6 +121,9 @@ cleanup_after_successful_rebase_and_exit() {
     echo "Press Enter to continue after fixing build:"
     read -r
     regenerate_changes_diff "origin/main"
+    echo "Generating contents of commit message into commit-message.txt."
```
Could you include the standard subject, so it'd be easier to have in the commit message.

`Update ANGLE to $(git log -1 ${COMMIT_HASH} --format=%cs) (${COMMIT_HASH}))
`    

Could you format the commit message body as:

```
Contains upstream commits:
git log --oneline ${PREVIOUS_ANGLE_COMMIT_HASH}..${COMMIT_HASH} --pretty="%h %s"
$(git log --oneline ${PREVIOUS_ANGLE_COMMIT_HASH}..${COMMIT_HASH} --pretty="%h %s")

```

E.g. the git log command would be included, so it's easy to cut-paste when verifying.

kenrussell replied:
Yes, certainly - done.

kkinnunen-apple reviewed (approved):
Too bad you had to do this. I started doing similar thing few times, but ended up in too ambitious changes that spaghettified themselves

zdobersek committed:
[Soup] USE(SOUP2) build fix after 250730@main
https://bugs.webkit.org/show_bug.cgi?id=240943

Patch by Žan Doberšek <zdobersek@igalia.com> on 2022-05-25
Unreviewed build fix for USE(SOUP2), switching to string literals in a couple of
places where URL::protocolIs() is expecting them.

* Source/WebKit/NetworkProcess/soup/NetworkDataTaskSoup.cpp:
(WebKit::NetworkDataTaskSoup::authenticateCallback):
(WebKit::NetworkDataTaskSoup::didStartRequest):

Canonical link: https://commits.webkit.org/251001@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294871 268f45cc-cd09-0410-ab3c-d52691b4dbfc

kenrussell committed:
Generate better ChangeLogs in update-angle script
https://bugs.webkit.org/show_bug.cgi?id=238649

Patch by Kenneth Russell <kbr@chromium.org> on 2022-05-25
Auto-generate the preferred commit message format during ANGLE rolls
into a commit-message.txt file.

Reviewed by Kimmo Kinnunen.

* Tools/Scripts/update-angle:

Canonical link: https://commits.webkit.org/251002@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@294872 268f45cc-cd09-0410-ab3c-d52691b4dbfc
